cpp/lib/include/RegexMatcher.h

```diff
@@ -83,6 +83,7 @@ class ThreadSafeRegexMatcher {
     MatchResult match(const std::string &subject, const size_t subject_start_offset = 0,
                       size_t * const start_pos = nullptr, size_t * const end_pos = nullptr) const;
     std::string replaceAll(const std::string &subject, const std::string &replacement) const;
+    std::string replaceWithBackreferences(const std::string &subject, const std::string &replacement, const bool global = false);
```

Comment:
I think it might be a good idea to add an interface documentation here, including a simple example usage.